geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHash.java

```diff
@@ -59,17 +62,31 @@ public RedisHash() {
     // for serialization
   }
 
+
   /**
    * Since GII (getInitialImage) can come in and call toData while other threads are modifying this
    * object, the striped executor will not protect toData. So any methods that modify "hash" needs
    * to be thread safe with toData.
    */
   @Override
-  public synchronized void toData(DataOutput out) throws IOException {
-    super.toData(out);
+  public void toData(DataOutput out, SerializationContext context) throws IOException {
```

Comment:
It looks like you removed synchronization on this method? Was that intended?

Reply:
merge conflict! apologies